src/instructlab/rag/process_docs.py

```diff
@@ -0,0 +1,108 @@
+# Standard
+from pathlib import Path
+from typing import Iterable
+import json
+import logging
+import os
+import tempfile
+import time
+
+# Third Party
+from docling.datamodel.base_models import ConversionStatus  # type: ignore
+from docling.datamodel.document import ConversionResult  # type: ignore
+from docling.document_converter import DocumentConverter  # type: ignore
+
+# First Party
+from instructlab.rag.taxonomy_utils import lookup_knowledge_files
+from instructlab.utils import clear_directory
+
+logger = logging.getLogger(__name__)
+
+
+def process_docs_from_taxonomy(taxonomy_path, taxonomy_base, output_dir):
+    # Taxonomy navigation strategy:
+    # Create temp folder that is deleted when the function returns
+    # Read taxonomy using read_taxonomy_leaf_nodes from instructlab-sdg package
+    # The above step also downloads the reference documents.
+    # Move all the downloaded documents under the temp folder
+    # Pass the list to process_docs_from_folder
+    with tempfile.TemporaryDirectory() as temp_dir:
+        logger.info(f"Temporary directory created: {temp_dir}")
+        knowledge_files = lookup_knowledge_files(taxonomy_path, taxonomy_base, temp_dir)
+        logger.info(f"Found {len(knowledge_files)} knowledge files")
+        logger.info(f"{knowledge_files}")
+
+        process_docs_from_folder(temp_dir, output_dir)
+
+
+def process_docs_from_folder(input_dir, output_dir):
+    """
+    Process user documents from a given `input_dir` folder to the given `output_dir` folder, using docling converters.
+    Latest version of docling schema is used (currently, v2).
+    """
+    logger.info(f"Processing {input_dir} to {output_dir}")
+
+    clear_directory(Path(output_dir))
+
+    source_files = _load_source_files(input_dir=input_dir)
+    logger.info(f"Transforming source files {[p.name for p in source_files]}")
+
+    doc_converter = DocumentConverter()
```

Comment:
When we use docling in SDG, we have some logic to automatically detect which OCR engine to use and configure Docling appropriately. You will likely need similar steps here, as Docling by default attempts to use EasyOCR which downloads model weights from GitHub behind the scenes but some downstream use-cases use alternative OCR implementations and should not reach out to download blobs from GitHub while converting documents.

See https://github.com/instructlab/sdg/blob/v0.6.3/src/instructlab/sdg/utils/chunkers.py#L261-L270 for how we do that today.

Reply:
Sounds good.  I will make that change for OCR.

I see earlier in the same code it also uses `docling_model_path`.  @bbrowning or @khaledsulayman do either of you know what's going on with `docling_model_path`?  From what I can tell, it is currently [hard-coded to None](https://github.com/instructlab/sdg/blob/725e5ab810900ce83b53d63ca9b2ebda42c2bf56/src/instructlab/sdg/generate_data.py#L211C1-L211C30) in `_sdg_init`, and I can't find any code path that sets it to anything else.  So I guess it is just a placeholder for an actual configuration option to come in the future.  Is that right or am I missing something?

Reply:
It's not hardcoded to None - that's just the default value before it goes into the logic a bit further down in that file to locate the docling models. This is how we find the docling models to use, in scenarios (like downstream) where we cannot download these on the fly from the network and instead they need to be shipped on-disk. It's probably worth syncing with @khaledsulayman on how we handle these docling nuances for downstream needs, where the environments are more restricted than upstream and we should not be reaching out to the internet.

Reply:
Oh, I see now; sorry I missed that in the first review.  I will DM @khaledsulayman to try to get more clarity.